single.php

```diff
@@ -9,20 +9,62 @@
  * @since    Timber 0.1
  */
 
+// Add custom css class for body element
+function add_body_classes_for_post( $classes ) {
+	$classes[] = 'post_body';
+	return $classes;
+}
+add_filter( 'body_class', 'add_body_classes_for_post' );
+
+
+// Initializing variables.
+$context                     = Timber::get_context();
+$post                        = Timber::query_post();
+$context['post']             = $post;
 
-$context = Timber::get_context();
-$post    = Timber::query_post();
 
-$page_meta_data             = get_post_meta( $post->ID );
-$articles_title             = $page_meta_data['p4_articles_title'][0];
-$articles_count             = intval( $page_meta_data['p4_articles_count'][0] );
-$articles_count             = $articles_count == 0 ? 3 : $articles_count;
-$context['author_override'] = $page_meta_data['p4_author_override'][0];
+// Get the cmb2 custom fields data
+// Articles block parameters to populate the articles block
+// Author override parameter. If this is set then the author profile section will not be displayed
+$page_meta_data              = get_post_meta( $post->ID );
+$articles_title              = $page_meta_data['p4_articles_title'][0];
+$articles_count              = intval( $page_meta_data['p4_articles_count'][0] );
+$articles_count              = $articles_count == 0 ? 3 : $articles_count;
+$context['author_override']  = $page_meta_data['p4_author_override'][0];
+$context['background_image'] = $page_meta_data['p4_background_image_override'][0];
 
+
+// Build the shortcode for articles block
 if ( ! empty( $articles_title ) ) {
-	$post->articles = do_shortcode( "[shortcake_articles article_heading='$articles_title' article_count='$articles_count' /]" );
+	$post->articles = "[shortcake_articles article_heading='$articles_title' article_count='$articles_count' /]";
 }
-$context['post'] = $post;
+
+
+// Breaking the content to retrieve first 2 paragraphs
+$parts = preg_split("/(\r\n|\n|\r)/", $post->post_content, 3);
+if ( count( $parts ) == 3 ) {
```

Comment:
Same here, strict comparison.
Also, for forward compatibility with php 7.2, type cast $parts to (array) $parts to avoid Warning that count()
accepts only array or Object that implements Countable interface.